provider/pdns.go

```diff
@@ -57,6 +62,68 @@ const (
 	retryAfterTime = 250 * time.Millisecond
 )
 
+// PDNSConfig is comprised of the fields necessary to create a new PDNSProvider
+type PDNSConfig struct {
+	DomainFilter DomainFilter
+	DryRun       bool
+	Server       string
+	APIKey       string
+	TLSConfig    TLSConfig
+}
+
+// TLSConfig is comprised of the TLS-related fields necessary to create a new PDNSProvider
+type TLSConfig struct {
+	TLSEnabled            bool
+	CAFilePath            string
+	ClientCertFilePath    string
+	ClientCertKeyFilePath string
+}
+
+func (tlsConfig *TLSConfig) setHTTPClient(pdnsClientConfig *pgo.Configuration) error {
+	if !tlsConfig.TLSEnabled {
```

Comment:
Maybe we can make this simpler, just checking if TLS is enabled and not checking CAFilePath, ClientCertFilePath and ClientCertKeyFilePath?

You could also log.Debug to say something like skipping TLS for PDN Provider.

Reply:
I like your approach better - will make that change